File: Collector/OLD/LegacyJava/Collector/Dimension/BlockMaterials.java
package Collector.Dimension;

import com.badlogic.gdx.graphics.Texture;

import java.util.HashMap;

public class BlockMaterials {
    public static HashMap<String,Block> materials = new HashMap<>();
    public static HashMap<String,Texture> textures = new HashMap<>();

    //Private so the singleton can't be instantiated
    private BlockMaterials() {}

    public static void create(){
        materials.put("grass",new Block("grass"));
        materials.put("wood",new Block("wood"));
        materials.put("water",new Block("water"));
        materials.put("stone",new Block("stone"));
        materials.put("snow",new Block("snow"));
        materials.put("sand",new Block("sand"));
        materials.put("air",new Block("air"));
        materials.put("roof",new Block("roof"));
        materials.put("wall",new Block("wall"));

        for (String s:BlockMaterials.materials.keySet()) {
            textures.put(s,new Texture("assets/" + s + ".png"));
        }
    }
}

File: Collector/src/Dimension/BlockMaterials.cs
using System.Collections.Generic;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Content;
using Microsoft.Xna.Framework.Graphics;

namespace Collector.Dimension
{
    public static class BlockMaterials {
        public static readonly Dictionary<string,Block> Materials = new Dictionary<string, Block>();
        public static readonly Dictionary<string,Texture2D> Textures = new Dictionary<string, Texture2D>();

        //Private so the singleton can't be instantiated
        static BlockMaterials() {}
    
        public static void Initialize(ContentManager content){
            Materials.Add("grass",new Block("grass"));
            Materials.Add("wood",new Block("wood"));
            Materials.Add("water",new Block("water"));
            Materials.Add("stone",new Block("stone"));
            Materials.Add("snow",new Block("snow"));
            Materials.Add("sand",new Block("sand"));
            Materials.Add("air",new Block("air"));
            Materials.Add("roof",new Block("roof"));
            Materials.Add("wall",new Block("wall"));

            foreach (var name in Materials.Keys)
            {
                Textures.Add(name,content.Load<Texture2D>(name));
            }
        }
    }
}
